example-app/src/pages/PatentsPage.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { getPatents } from '../actions';
+import { RootState } from '../reducers/rootReducer';
+import { PatentsReducerState } from '../reducers/patentsReducer';
+import { PatentItem } from '../components/PatentItem/PatentItem';
+import { PatentDataShape } from '../services/NasaService';
+import { ErrorMessage } from '../components/ErrorMessage/ErrorMessage';
+
+export const PatentsPage: React.FC = () => {
+  const dispatch = useDispatch();
+  const { patentsData, error } =
+    useSelector<RootState, PatentsReducerState>(state => state.patentsReducer);
+
+  useEffect(() => {
+    if (patentsData.length || error) {
+      return;
```

Comment:
do not use early return in callbacks like this

apply a reversed approach, add the condition for `dispatch` invocation.